model/change_point_detection.go

```diff
@@ -328,6 +341,94 @@ func ReplaceChangePoints(ctx context.Context, env cedar.Environment, performance
 	return catcher.Resolve()
 }
 
+func GetTotalPagesForChangePointsGroupedByVersion(ctx context.Context, env cedar.Environment, projectId string, pageSize int) (int, error) {
+	type CountResult struct {
+		Count int `bson:"count"`
+	}
+	countKey := "count"
+	pipe := []bson.M{
+		{
+			"$count": countKey,
+		},
+	}
+	pipe = append(getMatchAndUnwindForChangePointsGroupedByVersion(projectId), pipe...)
+	cur, err := env.GetDB().Collection(perfResultCollection).Aggregate(ctx, pipe)
+	if err != nil {
+		return 0, errors.Wrap(err, "Cannot aggregate to get count of change points grouped by version")
+	}
+	defer cur.Close(ctx)
+	var res CountResult
+	if cur.Next(ctx) {
+		err = cur.Decode(&res)
+		if err != nil {
+			return 0, errors.Wrap(err, "Cannot decode response of getting count of change points grouped by version")
+		}
+		return int(math.Ceil(float64(res.Count) / float64(pageSize))), nil
+	}
+	return 0, errors.New("Not able to get count of total changepoints matching query")
+}
+
+func getMatchAndUnwindForChangePointsGroupedByVersion(projectId string) []bson.M {
+	return []bson.M{
+		{
+			"$match": bson.M{
+				bsonutil.GetDottedKeyName(perfInfoKey, perfResultInfoProjectKey):             projectId,
+				bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, "0"): bson.M{"$exists": true},
```

Comment:
I guess this is more efficient than `{$ne:[]}`?

Reply:
Looks like it's about the same - when I do an explain plan, the one in the code is less than 100 ms faster

Reply:
ms or ns? If ms that seems pretty long from a user perspective, and we might want to look into the indexing?

In any case, I'll leave it up to you; I just prefer `{$exists: true, $ne:[]}` since aesthetically it translates better to "is set, and isn't empty", but may just be me.